x/programs/runtime/runtime.go

```diff
@@ -23,11 +26,12 @@ const (
 	deallocFnName = "dealloc"
 )
 
-func New(log logging.Logger, meter Meter, storage Storage) *runtime {
+func New(log logging.Logger, meter Meter, db database.Database, callerAddress ed25519.PublicKey) *runtime {
```

Comment:
having runtime take a database interface here I think is pretty limiting. The storage interface was meant as a way to abstract the backend. It is certainly needs work but I still think runtime should not care about what kind of storage it uses.